### ISSUE TITLE
`nvim_win_get_config` returns table value for `row` and `col` instead of number

### ISSUE THREAD

**Issue by mtbrock** [neovim/neovim#24430]
### Problem

`vim.api.nvim_win_get_config` returns table values for `row` and `col`:
```lua
{
  anchor = "NW",
  col = {
    [false] = 10,
    [true] = 3
  },
  external = false,
  focusable = true,
  height = 20,
  relative = "editor",
  row = {
    [false] = 10,
    [true] = 3
  },
  width = 20,
  zindex = 50
}
```

You have to access `row[false]` and `col[false]` to get the real value.

### Steps to reproduce

Write to file:
```lua
-- config_test.lua
local bid = vim.api.nvim_create_buf(true, false)
local wid = vim.api.nvim_open_win(bid, true, {
    relative = 'editor',
    width = 20,
    height = 20,
    row = 10,
    col = 10,
})

local config = vim.api.nvim_win_get_config(wid)
print(vim.inspect(config))
```

Run the file:
```sh
nvim -u NORC -c 'luafile %' config_test.lua
```

### Expected behavior

I expect `row` and `col` to be numbers, not tables.

### Neovim version (nvim -v)

0.9.1

### Vim (not Nvim) behaves the same?

n/a

### Operating system/version

Arch Linux (kernel version 6.4.4-arch1-1)

### Terminal name/version

kitty 0.28.1

### $TERM environment variable

tmux-256color

### Installation

Arch Extra repo

**Comment by clason** [comment]
Looks like an issue with marshaling the dicts over the vimscript->lua bridge (0 -> false, 1 -> true)

**Comment by seandewar** [comment]
The `row` and `col` values are fractional, but Lua numbers with integer values are normally converted to Vim script numbers (integers). The `[true]=3` field (`[vim.type_idx]=vim.types.float`) is there to preserve the fact that it should be converted to a float.

**Comment by zeertzjq** [comment]
I don't think this behavior should be changed. The same behavior also applies to other functions like `nvim_eval` and `nvim_get_var`, where it is actually needed to distinguish a `Float` from a `Number`, and there is no reason to have a special case for `nvim_win_get_config`.

**Comment by mikesmithgh** [comment]
Thank you for the comments. It took me a couple times of reading this and the docs to understand what is going on so I'll put my understanding here in case it helps anyone else who comes across this in the future.

Given the result:
```lua
  col = {
    [false] = 10,
    [true] = 3
  },
```

Reading help example `:h lua-special-tbl`
```vim
{[vim.type_idx]=vim.types.float, [vim.val_idx]=1}` is converted to
     a floating-point 1.0. 
```

This looks confusing at first, but those are just variables.
- `:=vim.type_idx` = `true`
- `:=vim.types.float` = `3`
- `:=vim.val_idx` = `false`

So, given a lua table with two elements that have the keys `true` and `false`, this is recognized as the special table with a value and type.

Back to the example,
```lua
  col = {              col = {                    col = {               
    [false] = 10,  ->    [vim.val_idx] = 10,  ->    [vim.val_idx] = 10,  
    [true] = 3     ->    [vim.type_idx] = 3   ->    [vim.type_idx] = vim.types.float   
  },                   },                         },                    
```

col is a special table, with a value of 10 and a type of float.

**Comment by roycrippen4** [comment]
For anyone else ending up here like I did. If you need the row, col values from `vim.api.nvim_win_get_config(winnr)` you would do the following:

```lua
local config = vim.api.nvim_win_get_config(winnr)
local row = config.row[vim.val_idx]
local col = config.col[vim.val_idx]
```

**Comment by zeertzjq** [comment]
Fixed by #27284
